lib/generator.ml

```diff
@@ -15,23 +23,34 @@ let process_int_type schema =
   | _ -> failwith "int has unextected format"
 
 let get_ref_name ref =
-  match String.split_on_char '/' ref with
+  let uri, pointer =
+    match String.split_on_char '#' ref with
+    | [ uri; pointer ] -> uri, Some pointer
+    | [ uri ] -> uri, None
+    | _ -> failwith (sprintf "Unsupported remote ref value: %s. The URI contains multiple '#'." ref)
+  in
+  let name_of_path path =
+    match path |> String.split_on_char '/' |> List.rev |> List.hd with
+    | exception _ -> failwith (sprintf "Unsupported ref value: %s" ref)
+    | name -> name
+  in
+  match pointer with
+  | None -> name_of_path uri
+  | Some pointer ->
+  match String.split_on_char '/' pointer with
```

Comment:
I'm guessing this is not completely correct. If the pointer is to a subschema then the name shouldn't be just the last part of the pointer